src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.tagName}, id: ${element.id}.`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function creates a new promise for each event listener. While this works, it could lead to multiple promises being created if the same event occurs multiple times. Consider whether you need to handle multiple events or just the first occurrence.